packages/core/src/payment/strategies/worldpayaccess/worldpayaccess-payment-strategy.ts

```diff
@@ -9,97 +9,122 @@ import { WorldpayAccess3DSOptions, WorldpayAccessAdditionalAction, WorldpayAcces
     private async _processAdditionalAction(error: unknown, payment: OrderPaymentRequestBody): Promise<InternalCheckoutSelectors> {
-        if (!(error instanceof RequestError) || !some(error.body.errors, {code: 'additional_action_required'})) {
+        if (!(error instanceof RequestError) || !some(error.body.errors, { code: 'additional_action_required' })) {
             return Promise.reject(error);
         }
 
-        const iframe = this._createHiddenIframe(error.body);
-        return new Promise((resolve, reject) => {
-            const message_event = async (event: MessageEvent) => {
-                window.removeEventListener('message', message_event);
 
-                if (iframe) {
-                    iframe.remove();
+        return new Promise((resolve, reject) => {
+             const messageEvent = async (event: MessageEvent) => {
+                if (typeof event.data !== 'string' || !this._isValidJson(event.data)) {
+                    return reject(new Error(PAYMENT_CANNOT_CONTINUE));
                 }
 
-                if (typeof event.data !== 'string' || event.data === INVALID_URL) {
-                    return reject(new Error('Payment cannot continue'));
-                }
+                window.removeEventListener('message', messageEvent);
+                window.removeEventListener('remove_event_message', removeEvent);
+                iframeHidden.remove();
 
                 const data = JSON.parse(event.data);
                 const paymentPayload = merge({}, payment, { paymentData: { threeDSecure: { token: data.SessionId } } });
 
                 try {
-                    if (this._shouldRenderHostedForm) {
-                        await this._hostedForm?.submit(paymentPayload);
-                        return resolve(this._store.getState());
-                    }
-                    return resolve(await this._store.dispatch(this._paymentActionCreator.submitPayment( paymentPayload )));
+                    resolve(await submit(paymentPayload));
                 } catch (error) {
-                    if (!(error instanceof RequestError) || !some(error.body.errors, {code: 'three_d_secure_required'})) {
+                    if (!(error instanceof RequestError) || !some(error.body.errors, { code: 'three_d_secure_required' })) {
                         return reject(error);
                     }
 
-                    this._3dsrequired(error.body.three_ds_result, reject);
+                    if (!this._initializeOptions) {
+                        return reject(new NotInitializedError(
+                            NotInitializedErrorType.PaymentNotInitialized
+                        ));
+                    }
+
+                    const { onLoad } = this._initializeOptions;
+                    const frame = this._createIframe(error.body.three_ds_result);
+
+                    try {
+                        onLoad(frame, () => reject(new Error('Payment was cancelled')));
+                    } catch (e) {
```

Comment:
Do we need the `e` here, seems like we don't use it?

Reply:
it's correct, I don't need it but it's required by typescript.
![image](https://user-images.githubusercontent.com/4907027/190223602-04e5be74-5eb1-45c5-a701-d29ad300cae2.png)